release/src/main/scripts/download_github_actions_artifacts.py

```diff
@@ -266,28 +266,36 @@ def download_artifacts(run_id, repo_url, artifacts_dir, github_token):
     name = safe_get(artifact, "name")
     size_in_bytes = safe_get(artifact, "size_in_bytes")
 
-    download_single_artifact(
-        url, name, size_in_bytes, artifacts_dir, github_token)
+    fd, temp_file_path = tempfile.mkstemp(prefix=name, suffix=".zip")
+    try:
+      os.close(fd)
```

Comment:
This code will run on Py3 only so you can use tempfile.TemporaryDirectory() context manager:
```
    with tempfile.TemporaryDirectory() as tmp:
      temp_file_path = os.path.join(tmp, name)    # or name+'.zip'?
      download_single_artifact(
           url, name, size_in_bytes, temp_file_path, github_token)
      extract_single_artifact(temp_file_path, artifacts_dir)
```

Reply:
Great tip. I will use `TemporaryDirectory`, much more nicer.